### PR TITLE
Remove dead code in image promoter

### PR DESCRIPTION
**Pull request by jonjohnsonjr**
This isn't called by anything other than tests.

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Removes some codepaths that are no longer executed by anything (other than tests).

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

<3

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: internal/legacy/dockerregistry/inventory_test.go [hunk 1]
```diff
@@ -835,107 +835,6 @@ func TestSplitByKnownRegistries(t *testing.T) {
 	}
 }
 
-func TestCommandGeneration(t *testing.T) {
```
Comment (puerco) [inline internal/legacy/dockerregistry/inventory_test.go:838]: shouldn't we completely remove the invocations to the functions this test is checking before removing the test completely?

Comment (puerco) [reply internal/legacy/dockerregistry/inventory_test.go:838]: I think its only GetDeleteCmd

Comment (jonjohnsonjr) [reply internal/legacy/dockerregistry/inventory_test.go]: Done 👍

Comment (puerco) [reply internal/legacy/dockerregistry/inventory_test.go]: Thanks!

**Comment by puerco** [comment]
Thanks Jon!
/lgtm
/approve

**Comment by jonjohnsonjr** [comment]
@cpanato Anything left for that hold?

**Review by cpanato — APPROVED** [review]
thank you so much
/lgtm
/unhold

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**